### PR TITLE
add reply with typing + start typing

### PR DESCRIPTION
**Pull request by mikehmorrissey**
I felt that the bot's replies were pretty jarring / abrupt. I think introducing a slight delay and adding a typing indicator will provide a much smoother and natural user experience. 

I added a method to reply with typing delay based on how long the message is. I also added a method to simply start typing which is useful when calling out to third party apis. 

Let me know what you think.

### CODE REVIEW

File: lib/Slackbot_worker.js [hunk 1]
```diff
@@ -358,6 +358,28 @@ module.exports = function(botkit, config) {
         bot.say(msg,cb);
     };
 
+    bot.startTyping = function(src) {
```
Comment (anonrig) [inline lib/Slackbot_worker.js]: We started using Javascript Documentation found in http://usejsdoc.org.

Can you update all of your functions with necessary documentations?

File: lib/Slackbot_worker.js [hunk 2]
```diff
@@ -358,6 +358,28 @@ module.exports = function(botkit, config) {
...
+    bot.replyWithTyping = function(src, resp, cb) {
+        var text;
+
+        if (typeof(resp) == 'string') {
+            text = resp;
+        } else {
+            text = resp.text;
+        }
+
+        var typingLength = 500 / 60 * text.length;
```
Comment (anonrig) [inline lib/Slackbot_worker.js]: You've assumed every letter takes 8.33 miliseconds to type. For a sentence of 10 characters it takes 83.33 ms which is faster then most of the humans. In order to make it more human noticable, I suggest you increase your typingLength value.

File: lib/Slackbot_worker.js (hunk 1 above)
Comment (mikehmorrissey) [inline lib/Slackbot_worker.js]: @anonrig will do :+1:!

**Comment by mikehmorrissey** [comment]
@anonrig thanks for the feedback! made the changes.

**Comment by mateeyow** [comment]
@lunarmayor Is it possible to add the `typing` feature for converations like @MattieTK said.

**Comment by kevinsuh** [comment]
yes, this has not been added to conversations. would love to have a convo.sayWithTyping option

**Comment by maxhartshorn** [comment]
I agree this would be a natural fit into conversations

**Comment by mvaragnat** [comment]
I may be doing something wrong, but the "is typing" indicator doesn't disappear when the message is posted. it continues for a couple of seconds (before timing out ? I'm not sure).

**Comment by devhaoy** [comment]
Did anyone manage to make typing available in convo?  Pretty sure every one making a messenger bot needs this features

**Pull request merged**